File: engine/effects/handlers/condition.py
"""Condition effect handler scaffold."""

from __future__ import annotations

from engine.effects.base import Effect


def handle_condition(effect: Effect) -> dict:
    return {
        "status": "ok",
        "kind": "condition",
        "target": effect.target_unit_id,
        "payload": effect.payload,
    }

File: src/effects/handlers/condition.ts
/**
 * Condition effect handler scaffold.
 * Mirrors engine/effects/handlers/condition.py
 */

import { Effect } from "../base";

export function handleCondition(effect: Effect): Record<string, unknown> {
  return {
    status: "ok",
    kind: "condition",
    target: effect.targetUnitId,
    payload: effect.payload,
  };
}
